dotnet/src/Functions/Functions.OpenApi/Model/RestApiOperation.cs

```diff
@@ -238,7 +239,7 @@ private string BuildPath(string pathTemplate, IDictionary<string, object?> argum
             var node = OpenApiTypeConverter.Convert(parameter.Name, parameter.Type, argument);
 
             // Serializing the parameter and adding it to the path.
-            pathTemplate = pathTemplate.Replace($"{{{parameter.Name}}}", node.ToString().Trim('"'));
+            pathTemplate = pathTemplate.Replace($"{{{parameter.Name}}}", HttpUtility.UrlEncode(serializer.Invoke(parameter, node)));
```

Comment:
OpenAPI plugins use serializers to transform parameters and their values into a target format defined by parameter location and style. The serializers were designed to handle encoding in addition to transformation so that the rest api operation does not need to do it itself. It would be great if it were possible to maintain that separation between the operation and the serializers.

Reply:
@SergeyMenshykh I tried doing the encoding in the serializer but then the unit tests for adding HTTP headers all fail because the header values are encoded. Not clear we should be encoding these values hence I moved the encoding logic  here.

Reply:
Ah, got it. I think the fix is a good one for now. Later, we might change it if/when we need to support other path parameter styles.
<img width="842" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/85a78ccf-e5c2-45e3-b67d-c924bd69f5a2">